addons/purchase_stock/tests/test_reordering_rule.py

```diff
@@ -1602,3 +1602,51 @@ def test_orderpoint_warning_purchase_stock(self):
         self.product_01.seller_ids = False
         orderpoint.invalidate_recordset(fnames=['show_supply_warning'])
         self.assertTrue(orderpoint.show_supply_warning)
+
+    def test_orderpoint_warning_purchase_stock_multicompany(self):
+        """ Check that the supply warning computes correctly per company in a multi-company setup.
+
+        If a vendor exists only in one company, the reordering rule in that company
+        should not show a warning, while the rule in the other company without any
+        supplier should show a warning (warning icon).
+        """
+        self.env.user.group_ids |= self.env.ref('stock.group_stock_multi_locations')
+        company_a, company_b = self.env['res.company'].create([
+            {'name': 'Company A'},
+            {'name': 'Company B'},
+        ])
+        warehouse_a, warehouse_b = self.env['stock.warehouse'].search(
+            [('company_id', 'in', [company_a.id, company_b.id])], limit=2
+        ).sorted('company_id')
```

Comment:
```suggestion
        company_a, company_b = self.env['res.company'].create([{'name': 'Company A'}, {'name': 'Company B'}])
        warehouse_a, warehouse_b = self.env['stock.warehouse'].search([('company_id', 'in', [company_a.id, company_b.id])])
```
There should always be 2 warehouses in that company in the test case, unless you are not creating manual warehouses; remove the limit.

Reply:
Yes, there are only two `warehouses` utill we create new `warehouse` in that `company` so we can remove the `limit`,

I had used `sorted` to ensure that `warehouse_a` and `warehouse_b` don't mismatch with `company_a` and `company_b` based on `company_id`, but the search result already returns them in a correct order so we can remove it as well.

Thanks for the suggestion!